Client/qtTeamTalk/common.cpp

```diff
@@ -1105,36 +1105,21 @@ void addTextToSpeechMessage(TextToSpeechEvent event, const QString& msg)
 
 bool HostEntry::sameHost(const HostEntry& host, bool nickcheck) const
 {
-    if(nickcheck == true)
-    {
-        return ipaddr == host.ipaddr &&
-               tcpport == host.tcpport &&
-               udpport == host.udpport &&
-               encrypted == host.encrypted &&
-               /* srvpasswd == host.srvpasswd && */ //don't include passwords
-               username == host.username &&
-               /* password == host.password && */
-               nickname == host.nickname &&
-               channel == host.channel/* &&
-            hosts[i].chanpasswd == host.chanpasswd*/;
-    }
-    else
-    {
-        return ipaddr == host.ipaddr &&
-               tcpport == host.tcpport &&
-               udpport == host.udpport &&
-               encrypted == host.encrypted &&
-               /* srvpasswd == host.srvpasswd && */ //don't include passwords
-               username == host.username &&
-               /* password == host.password && */
-               channel == host.channel/* &&
-            hosts[i].chanpasswd == host.chanpasswd*/;
-    }
+    return ipaddr == host.ipaddr &&
+           tcpport == host.tcpport &&
+           udpport == host.udpport &&
+           encrypted == host.encrypted &&
+           /* srvpasswd == host.srvpasswd && */ //don't include passwords
+           username == host.username &&
+           /* password == host.password && */
+           (!nickcheck || nickname == host.nickname) &&
+           channel == host.channel
+           /* && hosts[i].chanpasswd == host.chanpasswd*/;
 }
 
-bool HostEntry::sameHostEntry(const HostEntry& host, bool nickcheck) const
+bool HostEntry::sameHostEntry(const HostEntry& host) const

```

Comment:
@Oreonan 'nickcheck' was not used in "sameHostEntry" so I removed it. In mainwindow.cpp you passed 'false' but it had no effect

Reply:
I just removed nickname check when serverlistdlg.cpp select a server. Now it's working as expected for me. It's OK for you?